models/web/snowplow_media_plays_by_pageview_this_run.sql

```diff
@@ -0,0 +1,112 @@
+{{
+  config(
+    materialized='table',
+    tags=["this_run"],
+    sort = 'start_tstamp',
+    dist = 'play_id'
+  )
+}}
+
+with prep as (
+
+  select
+    i.play_id,
+    i.page_view_id,
+    i.media_id,
+    i.domain_sessionid,
+    i.domain_userid,
+    max(i.duration) as duration,
+    i.media_type,
+    i.media_player_type,
+    i.page_referrer,
+    i.page_url,
+    max(i.source_url) as source_url,
+    i.geo_region_name,
+    i.br_name,
+    i.dvce_type,
+    i.os_name,
+    i.os_timezone,
+    min(start_tstamp) as start_tstamp,
+    max(start_tstamp) as end_tstamp,
+    sum(case when i.event_type = 'play' then 1 else 0 end) as plays,
+    sum(case when i.event_type in ('seek', 'seeked') then 1 else 0 end) as seeks,
+    sum(i.play_time_sec) as play_time_sec,
+    sum(i.play_time_sec_muted) as play_time_sec_muted,
+    coalesce(sum(i.playback_rate * i.play_time_sec) / nullif(sum(i.play_time_sec), 0), max(i.playback_rate)) as avg_playback_rate,
+    listagg(distinct percent_progress, ',') within group (order by percent_progress) as percent_progress_reached,
+    min(case when i.event_type in ('seek', 'seeked') then start_tstamp end) as first_seek_time,
+    max(i.percent_progress) as max_percent_progress
+
+  from  {{ ref('snowplow_media_interactions_this_run') }} as i
+
+  group by 1,2,3,4,5,7,8,9,10,12,13,14,15,16
+
+)
+
+, dedupe as (
+
+  select
+    *,
+    row_number() over (partition by page_view_id order by start_tstamp) as duplicate_count
+
+  from prep
+
+)
+
+, retention_rate as (
```

Comment:
Is it possible to calculate this with a case statement in the prep CTE?

Reply:
good point, despite the name it's a max calculation and does not have to come from the dedupe part so will be moved.